File: split/newpenalty.pas
function newpenalty(m: integer): halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 12;
  mem[p].hh.b1 := 0;
  mem[p + 1].int := m;
  newpenalty := p;
end;

File: cpp/src/newpenalty.cpp
#include "newpenalty.h"
#include "getnode.h"

halfword newpenalty(int m)
{
	auto p = getnode(2);
	type(p) = penalty_node;
	subtype(p) = 0;
	penalty(p) = m;
	return p;
}
